### PR TITLE
fix: handle negative JSON numbers

### PR DESCRIPTION
**Pull request by Vaibhav-C-S**
# Description

This PR fixes JSON log scanning for negative numeric values.

Previously, JSON numbers starting with - were not parsed through the JSON number path. Because of that, values like -123 and -12.5 could be emitted as KindUnknown instead of KindValue, and parsing of following JSON tokens could become inconsistent.
Before:
image
After:
image

## Type of change

- [x] Bug fix (fixes an issue)

## Changes

- Treat - as a valid start character for JSON numbers.
- Update readJSONNumber to consume an optional leading minus sign.
- Add tests for negative integer and negative decimal JSON values.

## Motivation

Negative numbers are valid JSON numeric values. They should be tokenized and colored consistently with positive JSON numbers.

For example:

{"count":-123}
{"temp":-12.5}

should parse -123 and -12.5 as KindValue.

### CODE REVIEW

File: scanner/logscan/logscan_test.go [hunk 1]
```diff
@@ -339,6 +339,48 @@ func TestScanner_tokens(t *testing.T) {
...
+				{Kind: KindValue, Text: "-1"},
+				{Kind: KindUnknown, Text: ","},
+				{Kind: KindValue, Text: "-2.5"},
+				{Kind: KindUnknown, Text: ","},
+				{Kind: KindValue, Text: "3"},
+				{Kind: KindParenthases, Text: "]"},
+				{Kind: KindParenthases, Text: "}"},
+				{Kind: KindNewline, Text: "\n"},
+			},
+		},
```
Comment (applejag) [inline scanner/logscan/logscan_test.go:383]: Could you add a test showing what happens on values with dashes between the digits? (Such as for a UUID)

File: scanner/logscan/logscan.go [hunk 2]
```diff
@@ -627,6 +627,16 @@ func readLetters(rest []byte) []byte {
 func readJSONNumber(rest []byte) []byte {
 	var index int
 	var hasDot bool
+
+	if len(rest) == 0 {
+		return nil
+	}
```
Comment (applejag) [inline scanner/logscan/logscan.go]: Why is this if statement needed?

**Review by applejag — CHANGES_REQUESTED** [review]
Looks good! Thanks!

Just some small comments

File: scanner/logscan/logscan_test.go (hunk 1 above)
Comment (Vaibhav-C-S) [reply scanner/logscan/logscan_test.go:383]: Sure, adding

File: scanner/logscan/logscan.go (hunk 2 above)
Comment (Vaibhav-C-S) [reply scanner/logscan/logscan.go]: Your are correct, looking back at the code, this check is redundant, removing it. Thanks.

**Comment by Vaibhav-C-S** [comment]
Hi @applejag, I’ve addressed the review comments. Please review when you get a chance. Thanks!

**Comment by Vaibhav-C-S** [comment]
Hi  @applejag , gentle ping, I’ve addressed the review comments and requested re-review. Please let me know if anything else is needed. Thanks

**Review by prune998 — APPROVED** [review]

**Comment by Vaibhav-C-S** [comment]
Hi @applejag, gentle follow-up on this. Please let me know if anything else is needed from my side. Thank you.

**Comment by applejag** [comment]
Oh I haven't responded to this. How uncivilized of me. Pardon

I'm currently on a long vacay lasting all of June and July. I can't promise to review this before start of August. I'm only sometimes on my laptop. I'm currently on my phone, and it sucks to review code on the phone.

Is it ok to wait with this until august?

**Comment by Vaibhav-C-S** [comment]
No worries at all, August is completely fine. Please enjoy your vacation :)

**Review by applejag — APPROVED** [review]

**Comment by applejag** [comment]
this is currently blocked by https://github.com/kubecolor/kubecolor/pull/351, but will get this merged soon :)
